### PR TITLE
Update error message for invalid ranges in const entries

### PR DESCRIPTION
**Pull request by amreshk**

### CODE REVIEW

File: control-plane/p4RuntimeSerializer.cpp [hunk 1]
```diff
@@ -1616,10 +1616,15 @@ class P4RuntimeEntriesConverter {
...
             big_int maxValue = (big_int(1) << keyWidth) - 1;
-            // These should be guaranteed by the frontend
-            BUG_CHECK(*start <= *end, "Invalid range with start greater than end");
-            BUG_CHECK(*end <= maxValue, "End of range is too large");
+            // NOTE: If end value is > max allowed for keyWidth, value gets
+            // wrapped around. A warning is issued in this case by the frontend
+            // earlier.
+            // For e.g. 16 bit key has a max value of 65535, Range of (1..65536)
+            // will be converted to (1..0) and will fail below check.
+            if (*start > *end)
```
Comment (mbudiu-vmw) [inline control-plane/p4RuntimeSerializer.cpp]: Maybe you can use an ErrorType argument for ::error.

Comment (amreshk) [reply control-plane/p4RuntimeSerializer.cpp]: Did try using Errortype initially, but the output for invalid range type adds  a `'..'`, 
`[--Werror=invalid] error: ..: Invalid`
vs current
` error:  Invalid range`

Comment (hesingh) [reply control-plane/p4RuntimeSerializer.cpp]: It is caused by not using an IR Node in the error message.   See what IR Node you can find in the method and included it in the error message.  Then the IR Node is printed as %1% and your vars are printed with %2% onwards.

**Comment by hesingh** [comment]
Anyone checking in another PR to p4c, could use these changes to change this PR's code.

```
diff --git a/control-plane/p4RuntimeSerializer.cpp b/control-plane/p4RuntimeSerializer.cpp
index e6e14a04..116c5c3e 100644
--- a/control-plane/p4RuntimeSerializer.cpp
+++ b/control-plane/p4RuntimeSerializer.cpp
@@ -1623,8 +1623,9 @@ class P4RuntimeEntriesConverter {
             // earlier.
             // For e.g. 16 bit key has a max value of 65535, Range of (1..65536)
             // will be converted to (1..0) and will fail below check.
-            if (*start > *end)
-                ::error("%s Invalid range for table entry", kr->srcInfo);
+           if (*start > *end)
+                ::error(ErrorType::ERR_INVALID, "%2% Invalid range for table entry",
+                        k, kr->srcInfo);
             if (*start == 0 && *end == maxValue)  // don't care
                 return;
             startStr = stringReprConstant(*start, keyWidth);

```

I unit tested the code changes by commenting out `if (*start > *end)` in p4c code.

```
hemant@ubuntu:~/p4c/build$ ./p4c-bm2-ss --std p4-16 ../testdata/p4_16_samples/table-entries-range-bmv2.p4 --p4runtime-files tmp.txt
../testdata/p4_16_samples/table-entries-range-bmv2.p4(64): [--Werror=invalid] error: ..: Invalid  Invalid range for table entry
            1..8 : a_with_control_params(21);
            ^^^^
../testdata/p4_16_samples/table-entries-range-bmv2.p4(64)
            1..8 : a_with_control_params(21);
            ^^^^
../testdata/p4_16_samples/table-entries-range-bmv2.p4(65): [--Werror=invalid] error: ..: Invalid  Invalid range for table entry
            6..12: a_with_control_params(22);
            ^^^^^
../testdata/p4_16_samples/table-entries-range-bmv2.p4(65)
            6..12: a_with_control_params(22);
            ^^^^^
hemant@ubuntu:~/p4c/build$ 

```

**Pull request merged**